### DISCUSSION TITLE
Is novelibre meant to be used in connection with document templates?

### DISCUSSION THREAD

**Discussion by fpolli** [peter88213/novelibre#27] [Q&A] [answered]
Is this something you would use with a template, such as the [Modern literature book](https://extensions.libreoffice.org/en/extensions/show/modern-literature-book) template for formatting of the manuscript?

**Comment by peter88213** [comment, marked as answer]
Yes, the final result is an .odt document that can be published using any template. Just try it yourself.

But remember that you only manage the book content with *novelibre*. You will have to provide your own frontmatter and backmatter, as is usual in books. 
One possible method would be to copy the content of the exported document into the example document you mentioned above. Sometimes book-on-demand service providers also offer templates that are adapted to their book formats. 

In another context, I provided my own 5x8 Inch book template:
https://github.com/peter88213/yw2html/tree/main/add-on

I myself use a DTP program and have programmed an exporter for it (novx_xtg). But with LibreOffice you can also create decent books for home use if you know your way around a bit.

**Comment by fpolli** [comment]
Thanks!
I see there are no installation instructions for Mac. I just tried using the terminal:

python3 novelibre_v4.7.0.pyzw
DEPRECATION WARNING: The system version of Tk is deprecated and may be removed in a future release. Please don't rely on it. Set TK_SILENCE_DEPRECATION=1 to suppress this warning.
sh: xdg-open: command not found
2024-07-29 11:21:41.037 Python[83173:73782998] WARNING: Secure coding is automatically enabled for restorable state! However, not on all supported macOS versions of this application. Opt-in to secure coding explicitly by implementing NSApplicationDelegate.applicationSupportsSecureRestorableState:.

**Reply by peter88213** [reply]
I can't say much about the Mac, because I don't have one. As far as I know, there are often outdated Python installations out of the box under OSx. The message you got indicates that your Tk/tkinter installation, which novelibre needs for its user interface, may not be up to date either. I recommend a web search on the topic “Update Python and tkinter on OSx”.

If I understand your message correctly, it's just a warning. It would be more interesting to know whether novelibre starts on your computer or not. 

In general, I would have preferred it if you had opened a separate thread for this problem in this discussion, so that other Mac users could also become aware of it and possibly help you further. Unfortunately, this discussion forum does not allow individual posts to be moved to other threads.

**Comment by fpolli** [comment]
Sorry about that. I will start a new thread on installing on Mac to chronicle my process for any future Mac users out there. I have a very new Mac so it should be good for a while. I do know that I have python 3.9, but I don't know about tkinter. I think likely not from an article I found, but I will pursue this later. The app looks like it will be very helpful when I start on volume II!

-------------
Fred Polli
Additional contact information available

**Reply by peter88213** [reply]
> I will start a new thread on installing on Mac to chronicle my process for any future Mac users out there. I have a very new Mac so it should be good for a while.

That would be great. If you succeed, I will set the Mac on the system requirements list. 

> I do know that I have python 3.9, but I don't know about tkinter. 

Python 3.9 is fine. *tkinter* is a Python library that connects the application to the *Tk* graphical user interface that comes with the Windows Python installation. Some Linux distributions don't provide *Tk* out of the box, so the users have to install the *python3-tk* package. According to my tkinter manual, *tkinter* and *Tk* should be included with the MacOS Python installation. 

Good luck.
